tests/src/Tables/ColumnTest.php

```diff
@@ -164,3 +164,34 @@
         ->assertTableSelectColumnHasOptions('with_options', ['red' => 'Red', 'blue' => 'Blue'], $post)
         ->assertTableSelectColumnDoesNotHaveOptions('with_options', ['one' => 'One', 'two' => 'Two'], $post);
 });
+
+it('can assert that a column exists with the given configuration', function () {
+    $post_published = Post::factory()->create([
+        'is_published' => true,
+    ]);
+
+    livewire(PostsTable::class)
+        ->assertTableColumnExists('title2', function (Filament\Tables\Columns\TextColumn $column) {
+            return $column->isSortable() &&
+                $column->isSearchable() &&
+                $column->getPrefix() == 'published';
+        }, $post_published);
+
+    $post_unpublished = Post::factory()->create([
+        'is_published' => false,
+    ]);
+
+    livewire(PostsTable::class)
+        ->assertTableColumnExists('title2', function (Filament\Tables\Columns\TextColumn $column) {
+            return $column->getPrefix() == 'unpublished';
+        }, $post_unpublished);
+
+    $this->expectException('PHPUnit\Framework\ExpectationFailedException');
+    $this->expectExceptionMessage('Failed asserting that a field with the name [title] and provided configuration exists on the [' . PostsTable::class . '] component');
+
+    livewire(PostsTable::class)
+        ->assertTableColumnExists('title', function (Filament\Tables\Columns\TextColumn $column) {
+            return $column->isTime();
+        }, $post_published);
+
+});
```

Comment:
Please use camelCase for variable names

Reply:
@danharrin Would you accept a PR for a custom PHPStan rule to pick up on these?

Reply:
Sure!